gmsh_utils/gmsh_IO.py

```diff
@@ -451,17 +455,42 @@ def generate_geometry(self, layer_parameters: Dict[str, Any], model_name: str):
 
         self.extract_geo_data()
 
+    def set_mesh_size_of_group(self, group_name: str, mesh_size=-1) -> None:
+        """
+        Sets the customized mesh size of a specific group
+
+        Args:
+             - group_name: The name of the group provided by user input
+             - mesh_size (float): The customized mesh size provided by user for a specific group.
+
+        Returns:
+            None
+        """
+
+        if mesh_size <= 0:
+            raise ValueError(f"The mesh size of {group_name} is smaller than or equal to zero. "
+                             f"Please provide a mesh size larger than zero.")
+        group = self.geo_data["physical_groups"][group_name]
+        geometry_ids = group["geometry_ids"]
+        for geometry_id in geometry_ids:
+            ndim = group["ndim"]
+            if ndim == 0:
+                gmsh.model.mesh.setSize([(ndim, geometry_id)], mesh_size)
+            else:
+                entities_list = gmsh.model.getBoundary([(ndim, geometry_id)], recursive=True)
+                gmsh.model.mesh.setSize(entities_list, mesh_size)
```

Comment:
I think the mesh_size should also be added to geo_data["physical_groups"]["group_name"].

Because now there are problems in gmsh_io.generate_mesh(), while using this function, the mesh size data is forgotten, because this function resets gmsh.

I think gmsh_io.generate_geo_from_geo_data() has to be extended by also readding the mesh size